src/database.cpp

```diff
@@ -204,6 +205,62 @@ namespace {
         }
     };
 
+    template <typename IntT, typename ArrayT>
+    class array_of_int_mapper : public abstract_mapper<std::vector<IntT>>, public direct_mapper<ArrayT>
+    {
+    public:
+        explicit array_of_int_mapper(const optional<string> &name, const mapper_factory &creator) :
+            abstract_mapper_base(name),
+            abstract_mapper<std::vector<IntT>>(name),
+            direct_mapper<ArrayT>(name, creator)
+        {}
+
+        virtual std::unique_ptr<cloneable>
+        clone_impl() const override {
+            return quince::make_unique<array_of_int_mapper<IntT,ArrayT>>(*this);
+        }
+
+        virtual void from_row(const row &src, std::vector<IntT> &dest) const override {
+            ArrayT text;
+            direct_mapper<ArrayT>::from_row(src, text);
+            std::stringstream ss(std::string(text).substr(1, std::string(text).size() -2));
+            while(!(ss.eof() || ss.bad() || ss.fail()))
+            {
+                IntT num;
+                ss >> num;
+                dest.push_back(num);
+                ss.ignore(1, ',');
+            }
```

Comment:
A check is required to determine if the empty string was returned indicating an empty array was returned.

```cpp
if (ss.str() != "" ) {
    IntT num;
    while (ss >> num) {
        dest.push_back(num);
        ss.ignore(1, ',');
    }
}
```